### PR TITLE
Make H3StaticTable class static

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/H3StaticTable.cs
Comment (Youssef1313) [inline src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/H3StaticTable.cs:50]: The indexers aren't allowed in static class. I tried to check if it was used any where but found none. However, I can't confirm because I run into build problems on my machine, so I'll wait for the CI build to finish and check.

**Comment by Tratcher** [comment]
This is going to require a matching PR in AspNetCore as well.
https://github.com/dotnet/runtime/blob/3688c871daeecc9a208f12eff0ae1a876bb6b947/src/libraries/Common/src/System/Net/Http/aspnetcore/ReadMe.SharedCode.md

**Review by jkotalik — APPROVED** [review]

**Comment by Youssef1313** [comment]
> 
> 
> This is going to require a matching PR in AspNetCore as well.
> https://github.com/dotnet/runtime/blob/3688c871daeecc9a208f12eff0ae1a876bb6b947/src/libraries/Common/src/System/Net/Http/aspnetcore/ReadMe.SharedCode.md

Isn't this done automatically, just like https://github.com/dotnet/aspnetcore/pull/20640?

**Comment by Tratcher** [comment]
> Isn't this done automatically, just like [dotnet/aspnetcore#20640](https://github.com/dotnet/aspnetcore/pull/20640)?

That's a fail-safe to make sure we don't miss anything. We want to know changes compile and work in both repos before merging them in either. That said, this one's probably OK since it didn't change any of the API surface area consumed by HttpClient or Kestrel. We can auto-merge it when the tests pass.

**Comment by Youssef1313** [comment]
> 
> 
> > Isn't this done automatically, just like [dotnet/aspnetcore#20640](https://github.com/dotnet/aspnetcore/pull/20640)?
> 
> That's a fail-safe to make sure we don't miss anything. We want to know changes compile and work in both repos before merging them in either. That said, this one's probably OK since it didn't change any of the API surface area consumed by HttpClient or Kestrel. We can auto-merge it when the tests pass.

Ok, I'm going to create a similar PR in aspnetcore

**Comment by Youssef1313** [comment]
After giving a look, it won't actually compile correctly in aspnetcore 😆 
The indexer wasn't used here, but was used there. That's the reason it existed.

Because indexers can't be used in static class, I'll have to make a method instead of the indexer. Should this method be included here also? (given that it's used only in aspnetnet repo but not here)

**Comment by Tratcher** [comment]
> Should this method be included here also?

Yes, the shared files should match exactly. There's instructions in the readme for copying the files. You can develop the fix in either repo, but make sure to copy it when you're done.

**Comment by Youssef1313** [comment]
> 
> 
> > Should this method be included here also?
> 
> Yes, the shared files should match exactly. There's instructions in the readme for copying the files. You can develop the fix in either repo, but make sure to copy it when you're done.

Thanks for guidance, I'll fix it soon.

**Comment by Youssef1313** [comment]
I opened https://github.com/dotnet/aspnetcore/pull/21705

File: src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/QPackDecoder.cs
Comment (Youssef1313) [inline src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/QPackDecoder.cs:416]: Is the `index <= Count` part is correct?
A valid index is between 0 and Count - 1 (unless there is something special about this case?)

**Review by halter73 — APPROVED** [review]

File: src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/H3StaticTable.cs
Comment (stephentoub) [inline src/libraries/Common/src/System/Net/Http/aspnetcore/Http3/QPack/H3StaticTable.cs:30]: These statics should now be renamed to have an s_ rather than just _ prefix

**Comment by Youssef1313** [comment]
@tratcher @stephentoub The parallel PR (https://github.com/dotnet/aspnetcore/pull/21705) is merged.

**Comment by stephentoub** [comment]
@Tratcher, should I merge this?  Not sure how the mirror works.

**Comment by Tratcher** [comment]
@stephentoub yes please. Ideally we merge the changes into both repos on the same day and then the mirror doesn't have to do anything. It's only there as a failsafe.

**Pull request merged**
